### PR TITLE
feat: move build-service tests to this repo

### PR DESCRIPTION
**Pull request by psturc**
### JIRA
[KFLUXDP-872](https://redhat.atlassian.net/browse/KFLUXDP-872)

### Why

As part of the Konflux CI test decentralization effort, this PR migrates all build-service E2E tests from the shared konflux-ci/e2e-tests repository into the build-service repository itself.

**Faster feedback loop**
Test changes ship in the same PR as the code they validate, eliminating the need for cross-repo coordination between build-service and e2e-tests.

**Clearer ownership** 
Build-service developers own and maintain their tests alongside the code, making it easier to understand, update, and review test changes in context.

**Reduced coupling**
The shared e2e-tests repo becomes a source of friction when multiple service teams need to land changes simultaneously. Decentralizing tests removes this bottleneck.

**Scalable pattern**
This follows the same decentralization approach planned for other services (integration-service, release-service, etc.), where each repo will own its E2E tests and consume only shared helper packages from e2e-tests.

### Summary
* Migrate E2E tests from konflux-ci/e2e-tests into test/e2e/ with a lean Go test framework (client wrappers for GitHub, GitLab, Forgejo, Kubernetes, Tekton, Quay)
* Add Tekton task and pipeline to run in-repo tests on a Kind cluster in CI
* Add Forgejo/Codeberg provider support (client, git adapter, provider registration)
* Fix parallel test stability: Quay private repo race condition, PaC controller OOM (memory patch to 512Mi), image manifest propagation retries, PipelineRun retrigger for Forgejo

### Changes
* test/e2e/ — full test framework and test suites (pac_build, multi_component, renovate, secret_lookup, annotations, image_source)
* integration-tests/ — Tekton task (build-service-e2e.yaml) and shell script with secret loading, Ginkgo execution, cluster tuning
* .tekton/ — pipeline references updated

### Fixes
* reduced excessive timeouts (30min→5min)
* ImageFromPipelineRun retry logic
* Forgejo retrigger in RetriggerComponentPipelineRun
* non-fatal Forgejo client init
* corrected Codeberg API base URL

### Differences from upstream e2e-tests (excluding import paths):
* Switched pipeline type from DockerBuild to DockerBuildOciTA
* Wrapped ImageFromPipelineRun calls in Eventually blocks to handle Quay propagation delays
* Reduced PipelineRun timeouts from 30min to 5min to fail faster
* Increased purge PR timeout from 1min to 5min for Forgejo
* Fixed typo in purge PR error message ("traget" → "target")
* Marked renovate tests as Pending (blocked on [STONEBLD-4620](https://redhat.atlassian.net/browse/STONEBLD-4620))
* Only migrated build-service-relevant tests - excluded build_templates and source_build tests

### Checklist:
 - PR has reference to the issue(s) it resolves
 - Write / update unit tests
 - Write / update integration (envtest) tests
 - Ensure there is an issue for e2e tests if needed
 - Ensure `make test` passes
 - Ensure test coverage hasn't decreased
 - Test code changes manually
 - Update readme and documentation
 - Write PR description that highlights overall changes and add usage examples if applicable

[KFLUXDP-872]: https://redhat.atlassian.net/browse/KFLUXDP-872?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[STONEBLD-4620]: https://redhat.atlassian.net/browse/STONEBLD-4620?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: e2e-tests/pipelines/konflux-e2e-tests.yaml [hunk 1]
```diff
@@ -160,16 +160,14 @@ spec:
         resolver: git
         params:
           - name: url
-            value: https://github.com/konflux-ci/e2e-tests.git
+            value: https://github.com/psturc/build-service.git
           - name: revision
-            value: main
+            value: KFLUXDP-872
```
Comment (psturc) [inline e2e-tests/pipelines/konflux-e2e-tests.yaml:154]: I think we need to keep this and after merge, create a follow-up PR to point to konflux-ci main

**Review by MartinBasti — APPROVED** [review]

File: test/e2e/pkg/clients/forgejo/git.go [hunk 2]
```diff
@@ -0,0 +1,372 @@
+package forgejo
```
Comment (tisutisu) [inline test/e2e/pkg/clients/forgejo/git.go]: Shall we keep common packages in e2e-tests repo itself, since it can be reused by other teams like `integration`?

Comment (flacatus) [reply test/e2e/pkg/clients/forgejo/git.go]: We can but this in the future can lead to hell deps if teams start tomuse different packages in their code deps

Comment (tisutisu) [reply test/e2e/pkg/clients/forgejo/git.go]: Even build components have multiple repos like build-service, image-controller, build-definitions, if we follow this approach and keep separate copies of the same package in different repos, it would be hard to manage, every change in the common packages need to be done in multiple repos.

File: e2e-tests/scripts/build-service-e2e.sh [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+timeout "$E2E_TIMEOUT" ginkgo \
+    -p \
+    --procs="${GINKGO_PROCS}" \
+    -v \
+    --no-color \
+    --output-interceptor-mode=none \
+    --timeout=90m \
+    --label-filter="${LABEL_FILTER}" \
+    --junit-report=e2e-report.xml \
+    --output-dir="${ARTIFACT_DIR}" \
```
Comment (tisutisu) [inline e2e-tests/scripts/build-service-e2e.sh:96]: Adding `--fail-on-empty` flag with this ginkgo command would be helpful.

File: test/e2e/pkg/clients/forgejo/git.go (hunk 2 above)
Comment (flacatus) [reply test/e2e/pkg/clients/forgejo/git.go]: @psturc wdyt about creating a repo named tests-common and move all there?

Comment (psturc) [reply test/e2e/pkg/clients/forgejo/git.go]: that could be an option, but also wondering if we could just keep them in e2e-tests repo?

File: test/e2e/tests/build/annotations.go [hunk 4]
```diff
@@ -0,0 +1,123 @@
+package build
```
Comment (tisutisu) [inline test/e2e/tests/build/annotations.go]: the directory structure for the test is little confusing, to help discover the tests for new users of the repository, from the root directory, it would be better to follow directory structure similar to the below
```
tests/
├── e2e
│   ├── annotations.go
│   └── pac_build.go
└── pkg
```

File: test/e2e/pkg/clients/forgejo/git.go (hunk 2 above)
Comment (flacatus) [reply test/e2e/pkg/clients/forgejo/git.go]: I think the problem is the dependency hell we have there. If we try to go get e2e tests here we can broke things and adding thousands of unnecesary deps

Comment (flacatus) [reply test/e2e/pkg/clients/forgejo/git.go]: oh wait, You are adding the own go.mod for the e2e- Can be an option after descentralization to rename e2e repo.

File: test/e2e/tests/build/annotations.go (hunk 4 above)
Comment (psturc) [reply test/e2e/tests/build/annotations.go]: ok did some changes locally, now I have a dir structure like this, wdyt
```
test
    ├── e2e
    │   ├── annotations.go
    │   ├── const.go
    │   ├── e2e_suite_test.go
    │   ├── git_provider_config.go
    │   ├── helpers.go
    │   ├── multi_component.go
    │   ├── pac_build.go
    │   ├── renovate.go
    │   └── secret_lookup.go
    ├── go.mod
    ├── go.sum
    └── pkg
        ├── clients
        │   ├── has
        │   │   ├── applications.go
        │   │   ├── components.go
        │   │   └── controller.go
        │   ├── imagecontroller
        │   │   ├── controller.go
        │   │   └── image_repository.go
        │   ├── integration
        │   │   └── controller.go
        │   ├── kube
        │   │   ├── client.go
        │   │   ├── controller.go
        │   │   ├── namespace.go
        │   │   ├── pod.go
        │   │   ├── secret.go
        │   │   └── service_account.go
        │   ├── release
        │   │   ├── controller.go
        │   │   ├── plans.go
        │   │   └── releases.go
        │   └── tekton
        │       ├── containers.go
        │       ├── controller.go
        │       ├── pipelineruns.go
        │       └── repository.go
        ├── constants
        │   └── constants.go
        ├── framework
        │   └── framework.go
        ├── logs
        │   └── logs.go
        └── utils
            ├── build
            │   ├── custom_bundle.go
            │   ├── git.go
            │   ├── image.go
            │   └── quay.go
            ├── retry_transport.go
            ├── tekton
            │   ├── persistent_volume_claims.go
            │   └── pipeline_ref.go
            └── util.go

**Review by flacatus — APPROVED** [review]

File: test/e2e/tests/build/annotations.go (hunk 4 above)
Comment (tisutisu) [reply test/e2e/tests/build/annotations.go]: looks good now.

**Comment by mmorhun** [comment]
@psturc could you add reasoning why this change is made?

**Comment by psturc** [comment]
@mmorhun please see the related jira ticket and feature for more details

**Comment by mmorhun** [comment]
I still don't see clear explanation why. Please update the PR description, not all people can access Jira.
Also, is it planned to decommission current e2e tests repository?

**Comment by flacatus** [comment]
@mmorhun Is related to https://redhat.atlassian.net/browse/KONFLUX-9276

The centralized repo(konflux-ci/e2e-tests) had become a bottleneck, every team depended on it, changes from one team could break another's tests, and it created a single point of failure in CI. By moving tests into the service repos, each team has full ownership of their test lifecycle...

The e2e tests repo remains like a package that contains global packages for all Konflux Teams if they want to use it

**Comment by flacatus** [comment]
@mmorhun ar we good here?

**Comment by mmorhun** [comment]
@flacatus we discussed it with the team and still have a few concerns:
1. Team owns several repositories which tests depend on common e2e tests framework base. Splitting tests in target repositories means duplicating parts of the test framework.
2. Some test scenarios test cross component interactions, for example: component + image repository. Duplicating such tests doesn't look good to me.
3. Since it's e2e test they verify cross component interactions, for example, one of our tests has scenario with Conforma to make sure the checks pass on the produced build. How such situation will be handled in the decentralized repos?

If we keep only build-service e2e tests here, it won't be Konflux e2e tests any more and we'll be missing some important use cases to check.
If we include combinations with other services we would duplicate test scenarios and duplicate framework which would noticeable increase maintenance burden.

**Comment by flacatus** [comment]
@mmorhun 

  **Common test framework duplication**. The plan is not to copy-paste the framework into every repo. The shared test framework (clients, utilities, helpers) will be extracted into a reusable  Go module that teams can import as a dependency. Each team pulls in what they need without duplicating code. The e2e-tests repo is being refactored to serve as that common library — teams  consume it, they don't fork it.

  **Cross-component interaction tests**. These are exactly the kind of tests that should not live in individual service repos. We distinguish between two types of tests:
  - **Component-level e2e tests** — tests that validate a specific service's behavior (e.g., PaC builds, annotations). These move to the build service repo.
  - **Conformance / integration tests** — tests that validate cross-component interactions across Konflux (e.g., build + image repository + Conforma checks). These stay centralized in konflux-ci/konflux-ci as the Konflux conformance suite abd this will run as part of infra-deployments repository

**Comment by mmorhun** [comment]
Thank you for the response, @flacatus, a few additional questions:

- According to the above, `component-level e2e tests` should test the behavior of corresponding Konflux component itself. However, to have e2e flow, other services should often be involved. Will it lead to duplication between component repos? Or should we have kind of integration tests instead e2e tests in component repos?
- Who will write / maintain the `conformance / integration tests`?
- Common test framework saves from duplication in the repos, but what about component-level (repo) tests and the conformance tests?

**Comment by flacatus** [comment]
@mmorhun let me clarify:

  **Component-level e2e tests and duplication**. You're right that even component-level tests often involve other services to get a full e2e flow. The key distinction is what you're asserting on. In build-service's repo, you test build-service behavior — even if that requires a running Integration Service or other components in the cluster. You're not testing that Integration Service works correctly, you're using it as part of the setup to validate your component. If two teams both need to trigger a build as part of their test setup, that's not duplication — they're testing different outcomes from the same trigger. Think of it as integration tests scoped to your component's responsibilities.

  **Who maintains the conformance suite**? The conformance suite in konflux-ci/konflux-ci will be maintained by the Vanguard team, but it's open for contribution from every team. If your team identifies a cross-component scenario that should be covered at the system level, you can contribute it there. This is the same model Kubernetes uses — SIG Testing maintains the conformance suite, but any **SIG** can contribute tests that validate cross-cutting behavior.

  Component-level tests vs conformance tests. These are complementary, not overlapping. Teams can also have their own conformance suite at the component level — a set of tests that define "this component is working correctly." The system-level conformance suite in konflux-ci/konflux-ci validates that all components work together as a whole. Again, this mirrors how Kubernetes works: each component has its own e2e tests, and then there's a separate conformance suite that validates the full cluster behavior. Both are valuable, and they serve different purposes.
  
  Edit: Teams are allow to setup their own Tests in Infra Deployments to run when their components get updated

File: test/pkg/framework/framework.go [hunk 5]
```diff
@@ -0,0 +1,184 @@
+package framework
```
Comment (mmorhun) [inline test/pkg/framework/framework.go]: Shouldn't the framework be in e2e tests repo according to the conversation above?

File: test/pkg/clients/has/applications.go [hunk 6]
```diff
@@ -0,0 +1,73 @@
+package has
```
Comment (tisutisu) [inline test/pkg/clients/has/applications.go]: > Each team pulls in what they need without duplicating code. The e2e-tests repo is being refactored to serve as that common library — teams consume it, they don't fork it.

Moving these dependent packages does not align with the discussion happened above.

Comment (flacatus) [reply test/pkg/clients/has/applications.go]: Yes, but as I mention in this slack/PR only libraries like Quay/Github/Gitlab stay in the e2e repo..

Comment (mmorhun) [reply test/pkg/clients/has/applications.go]: Sorry for my misunderstanding, I though the the framework is `framework` in the e2e tests code.
In such case, we'd need to duplicate part of the framework in build-service, image-controller, build-definitions, ... which is not great.
@flacatus I understand that QE team don't want to support that part of the framework, however, it's still beneficial to keep it in a common place for all teams. If possible, We'd prefer to keep the framework (including parts that operate on k8s objects, etc.) in the e2e tests repo, but maintained by appropriate teams (e.g. Component and ImageRepository becomes maintained by Build team, Snapshots by Integration team, ReleasePlan / ReleasePlanAdmission by Release team, etc.).
Does it make sense to you?

Comment (flacatus) [reply test/pkg/clients/has/applications.go]: Teams are already migrated to their own repo to use their controller APIs for the testing :(

Comment (flacatus) [reply test/pkg/clients/has/applications.go]: See https://github.com/konflux-ci/release-service/tree/main/e2e-tests

Comment (mmorhun) [reply test/pkg/clients/has/applications.go]: I see. It might make sense for teams that has only one service, so everything is in one place, but we need a common place. Can we keep and maintain our framework part in e2e tests repo?

Comment (flacatus) [reply test/pkg/clients/has/applications.go]: ok, Agree. Lets keep an use the framework from the e2e tests repo. @psturc Can we use has, integration release etc from e2e repo?

Comment (psturc) [reply test/pkg/clients/has/applications.go]: just to clarify, is there anything from the `test/pkg` directory (from this PR) that we want to keep or we want to use all the packages from e2e-tests repo and maintain only tests in build-service repo?

Comment (flacatus) [reply test/pkg/clients/has/applications.go]: @mmorhun can you clarify?

Comment (psturc) [reply test/pkg/clients/has/applications.go]: btw thoughts on how are we going to validate the changes to those packages in e2e-tests repo after tests are gone? ideally some tests should be triggered in the repo to do the validation of new changes.

Comment (flacatus) [reply test/pkg/clients/has/applications.go]: I would say conformance suite

Comment (tisutisu) [reply test/pkg/clients/has/applications.go]: > just to clarify, is there anything from the `test/pkg` directory (from this PR) that we want to keep or we want to use all the packages from e2e-tests repo and maintain only tests in build-service repo?

It would be better to keep all the packages (currently under `test/pkg` directory on this PR) in e2e-tests repository and only tests from `test/e2e` in build-service repo, to reduce duplicate across build repos.

File: integration-tests/scripts/build-service-e2e/0.1/build-service-e2e.sh [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+export CUSTOM_DOCKER_BUILD_OCI_TA_PIPELINE_BUNDLE="${CUSTOM_DOCKER_BUILD_OCI_TA_PIPELINE_BUNDLE:-quay.io/konflux-ci/tekton-catalog/pipeline-docker-build-oci-ta:devel}"
+log "INFO" "Using pipeline bundle: ${CUSTOM_DOCKER_BUILD_OCI_TA_PIPELINE_BUNDLE}"
+
+LABEL_FILTER="build-service"
+if [[ -n "${E2E_EXTRA_LABEL_FILTER:-}" ]]; then
+    LABEL_FILTER="${LABEL_FILTER} && ${E2E_EXTRA_LABEL_FILTER}"
+fi
+
+GINKGO_PROCS="${GINKGO_PROCS:-10}"
```
Comment (tisutisu) [inline integration-tests/scripts/build-service-e2e/0.1/build-service-e2e.sh]: Can we reduce this number to `5` ? rate limiting issues we are observing in CI can be avoided like below
```
Unexpected error:
    <*fmt.wrapError | 0xc0012d62e0>: 
    failed to create pull request: newIssue: NewIssue: "konflux-ci-qe-bot" posted 7 issues in under 10 minutes: rate limited
    {
        msg: "failed to create pull request: newIssue: NewIssue: \"konflux-ci-qe-bot\" posted 7 issues in under 10 minutes: rate limited",
        err: <*errors.errorString | 0xc0008821d0>{
            s: "newIssue: NewIssue: \"konflux-ci-qe-bot\" posted 7 issues in under 10 minutes: rate limited",
        },
    }
occurred
```

**Review by tisutisu — APPROVED** [review]

File: integration-tests/scripts/build-service-e2e/0.1/build-service-e2e.sh (hunk 7 above)
Comment (psturc) [reply integration-tests/scripts/build-service-e2e/0.1/build-service-e2e.sh]: done

File: test/pkg/framework/framework.go (hunk 5 above)
Comment (psturc) [reply test/pkg/framework/framework.go]: moved all back to e2e-tests repo

File: e2e-tests/tests/git_provider_config.go [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+
+const (
+	// E2E_GIT_PROVIDERS_ENV controls which git providers to run tests against.
+	// Comma-separated list of provider prefixes: "gh,gl,gt" or "gh" or "gl,gt"
+	// If not set or empty, all registered providers will be used.
+	// Examples:
+	//   E2E_GIT_PROVIDERS=gh           -> only GitHub
+	//   E2E_GIT_PROVIDERS=gl           -> only GitLab
+	//   E2E_GIT_PROVIDERS=gh,gl        -> GitHub and GitLab
+	//   E2E_GIT_PROVIDERS=gh,gl,gt     -> all three (GitHub, GitLab, Gitea)
```
Comment (mmorhun) [inline e2e-tests/tests/git_provider_config.go]: isn't it `Forgejo` (`fj`) instead of `Gitea` now?

File: e2e-tests/go.mod [hunk 9]
```diff
@@ -0,0 +1,275 @@
...
+	sigs.k8s.io/yaml v1.6.0 // indirect
+)
+
+replace github.com/cyphar/filepath-securejoin => github.com/cyphar/filepath-securejoin v0.3.6
+
+replace github.com/expr-lang/expr => github.com/expr-lang/expr v1.17.7
+
+replace github.com/go-chi/chi/v5 => github.com/go-chi/chi/v5 v5.2.4
+
+replace github.com/go-chi/chi => github.com/go-chi/chi/v5 v5.2.4
```
Comment (mmorhun) [inline e2e-tests/go.mod]: What's the point of having those separate from all other replacements?
Why do we need to replace deps?

**Review by mmorhun — COMMENTED** [review]
We have `test` and `integration-tests` in the repository root which probably should be under the same dir?

File: e2e-tests/tests/git_provider_config.go (hunk 8 above)
Comment (psturc) [reply e2e-tests/tests/git_provider_config.go]: that is copied from [here](https://github.com/konflux-ci/e2e-tests/blob/f0b222ad8cb697b5048301c521e32fd74b07e004/tests/build/git_provider_config.go#L26), I can rename it if needed

File: e2e-tests/go.mod (hunk 9 above)
Comment (psturc) [reply e2e-tests/go.mod]: also copied from [e2e-tests repo](https://github.com/konflux-ci/e2e-tests/blob/f0b222ad8cb697b5048301c521e32fd74b07e004/go.mod#L343-L349) - I will check if we can skip the replacements

Comment (mmorhun) [reply e2e-tests/go.mod]: Thank you!

File: e2e-tests/tests/git_provider_config.go (hunk 8 above)
Comment (mmorhun) [reply e2e-tests/tests/git_provider_config.go]: If it's not hard for you, but you can leave it as is, we are responsible for it...

File: e2e-tests/pipelines/konflux-e2e-tests-gitlab.yaml [hunk 10]
```diff
@@ -0,0 +1,280 @@
+---
```
Comment (mmorhun) [inline e2e-tests/pipelines/konflux-e2e-tests-gitlab.yaml:1]: I see that all 4 pipelines are almost duplicates.
Could you have common pipeline and separate small PipelineRuns?

File: e2e-tests/tests/const.go [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+	buildTemplatesKcpTestLabel       = "build-templates-kcp-e2e"
+	sourceBuildTestLabel             = "source-build-e2e"
+
+	helloWorldComponentGitSourceRepoName = "devfile-sample-hello-world"
+	helloWorldComponentDefaultBranch     = "default"
+	helloWorldComponentRevision          = "d2d03e69de912e3827c29b4c5b71ffe8bcb5dad8"
+
+	multiComponentGitSourceRepoName = "sample-multi-component"
+	multiComponentDefaultBranch     = "main"
+	multiComponentGitRevision       = "0d1835404efb8ab7bb1ab5b5b82cda1ebfda4b25"
```
Comment (tisutisu) [inline e2e-tests/tests/const.go]: @psturc can we incorporate [this change](https://github.com/konflux-ci/e2e-tests/pull/1862) here? it contains a fix related to component pipeline failed due to `OOMKilled` .

Comment (tisutisu) [reply e2e-tests/tests/const.go]: looks like there are many changes in e2e-tests repo and it would hard to pass CI, CI is running all the suites now.

File: e2e-tests/go.mod (hunk 9 above)
Comment (psturc) [reply e2e-tests/go.mod]: these replacements are here due to openshift/oc dependency. I created [this PR](https://github.com/konflux-ci/e2e-tests/pull/1864) to get rid of that dependency in build-service tests. so once that one is merged, the replacements should be no longer required

File: e2e-tests/tests/const.go (hunk 11 above)
Comment (psturc) [reply e2e-tests/tests/const.go]: merged the PR and incorporated the change into this PR

File: e2e-tests/tests/git_provider_config.go (hunk 8 above)
Comment (psturc) [reply e2e-tests/tests/git_provider_config.go]: updated

File: e2e-tests/README.md [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+### `pipelines/`
+
+A single parameterized Tekton Pipeline (`konflux-e2e-tests.yaml`) that provisions a Kind cluster on AWS, deploys Konflux, runs a subset of tests filtered by provider, collects artifacts, and deprovisions the cluster.
+
+The pipeline accepts two provider-specific parameters set by each IntegrationTestScenario:
+
+| IntegrationTestScenario | `provider-tag` | `e2e-label-filter` |
+|-------------------------|---------------|-------------------|
+| GitHub PaC build | `github-pac` | `github-webhook && github` |
+| GitHub integration | `github-integration` | `github && !github-webhook` |
```
Comment (mmorhun) [inline e2e-tests/README.md]: Do we really need to separate those?

**Review by mmorhun — APPROVED** [review]

File: e2e-tests/README.md (hunk 12 above)
Comment (psturc) [reply e2e-tests/README.md]: ok I reverted that, kept only 3 pipelines - github, gitlab, forgejo. let's see how it works

File: e2e-tests/pipelines/konflux-e2e-tests-gitlab.yaml (hunk 10 above)
Comment (psturc) [reply e2e-tests/pipelines/konflux-e2e-tests-gitlab.yaml:1]: done, 3 pipelines in the end

**Comment by tisutisu** [comment]
@psturc seeing only lint check is failed, can we please fix and merge this PR? since we changed ITS, till we merge this, other PR CI will continue to fail.

**Comment by psturc** [comment]
> @psturc seeing only lint check is failed, can we please fix and merge this PR? since we changed ITS, till we merge this, other PR CI will continue to fail.

fixed

File: e2e-tests/go.mod (hunk 9 above)
Comment (psturc) [reply e2e-tests/go.mod]: addressed in [this commit](https://github.com/konflux-ci/build-service/pull/586/commits/9d3dcad2ac388c625180d81af52172ae502e6c8b)
